File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function(nums, k) {
    let res = 0;
    let cnt = 0;
    let sum = 0;
    let n = nums.length, l = 0;
    for (let r = 0; r < n; ++r) {
        sum += nums[r];
        cnt++;
        while (sum * cnt >= k) {
            sum -= nums[l++];
            cnt--;
        }
        res += cnt;
    }
    return res;
};

File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.py
class Solution(object):
    def countSubarrays(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        res, cnt, sm = 0, 0, 0
        n, l = len(nums), 0
        for i in range (n):
            sm += nums[i]
            cnt += 1
            while sm * cnt >= k:
                cnt -= 1
                sm -= nums[l]
                l += 1
            res += cnt
        return res